crates/matrix-sdk/src/sliding_sync/list/mod.rs

```diff
@@ -20,14 +20,18 @@ use futures_core::Stream;
 use imbl::Vector;
 pub(super) use request_generator::*;
 pub use room_list_entry::RoomListEntry;
-use ruma::{api::client::sync::sync_events::v4, assign, events::StateEventType, OwnedRoomId, UInt};
+use ruma::{api::client::sync::sync_events::v4, assign, events::StateEventType, OwnedRoomId};
 use serde::{Deserialize, Serialize};
 use tokio::sync::mpsc::Sender;
 use tracing::{instrument, warn};
 
 use super::{Error, FrozenSlidingSyncRoom, SlidingSyncInternalMessage, SlidingSyncRoom};
 use crate::Result;
 
+/// The type used to express natural bounds (including but not limited to:
+/// ranges, timeline limit) in the sliding sync SDK.
+pub type Bound = u32;
```

Comment:
I wonder if we should name it `SlidingSyncBound` 🤔. Let's keep `Bound` as it's not exposed at the root of the crate _a priori_. It's public, but restricted to this module, so it's fine.